### PR TITLE
Feature/retrieve flights

### PR DESCRIPTION
**Pull request by NicoleHoerner**
US1: Retrieve and list flights

- in this user story to get started: purely API implementation/fetching data and display a few flights
- env.local file will be implemented later
- right now it is not possible to retrieve particular flights as API sends a limitation of 100 through which I can search/map. This data will be updated every time you refresh browser (if the particular flight is in this data of 100 you could display one particular flight)
- I styled the list a little bit as data would not have been readable nicely
- I limited the displayed flights to 10 at this point

**Please do NOT refresh browser page too many times as I have a monthly limited API request!**

### CODE REVIEW

File: pages/api/hello.js [hunk 1]
```diff
@@ -1,5 +1,10 @@
...
 
-export default function handler(req, res) {
-  res.status(200).json({ name: 'John Doe' })
+export default async function handler(req, res) {
+  const response = await fetch(
+    "http://api.aviationstack.com/v1/flights?access_key=eef146cc9441c581c88b1ba9363a5d4b"
+  );
+
+  const data = await response.json();
+  res.status(200).json({ data: data });
```
Comment (MerleMak) [inline pages/api/hello.js:9]: I think at this point there is no need to fetch in the api folder. 
For now it is reasonable to fetch in the frontend only, as some things are repeating themselves (for example response.json();  and data.data.data.

File: pages/index.js [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
 import { Inter } from "next/font/google";
+import { useEffect, useState } from "react";
 
 const inter = Inter({ subsets: ["latin"] });
 
 export default function Home() {
+  const [flights, setFlights] = useState([]);
+  useEffect(() => {
+    async function getFlights() {
+      const res = await fetch("/api/hello");
```
Comment (MerleMak) [inline pages/index.js:12]:
```suggestion
      const response = await fetch("http://api.aviationstack.com/v1/flights?access_key=eef146cc9441c581c88b1ba9363a5d4b");
```

File: pages/index.js [hunk 3]
```diff
@@ -1,10 +1,22 @@
...
+      const data = await res.json();
```
Comment (MerleMak) [inline pages/index.js:13]:
```suggestion
      const data = await response.json();
```

File: pages/index.js [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
+      setFlights(data.data.data.slice(0, 10));
+      console.log(data);
```
Comment (MerleMak) [inline pages/index.js:15]:
```suggestion
```

**Review by MerleMak — COMMENTED** [review]
Hi Nicole, I left you some comments for you to consider :)
Nice start, but some things are redundant. Keep up the good work! 
Good luck :))

File: pages/api/hello.js (hunk 1 above)
Comment (NicoleHoerner) [reply pages/api/hello.js:9]: Ok. It was suggested to put that function in the api folder from our coach. But I will change it!

**Comment by NicoleHoerner** [comment]
@MerleMak 
Hi there! I really tried implementing your suggested changes. But somehow it breaks my code and can't fix it. 
Can I leave it as it was?
What can/should I do? Because I desperate here!

**Review by Kheirah — APPROVED** [review]
The fetch within the API route is necessary since the aviation API does not allow https requests and the frontend enforces https connections. You're good to go to QA! =)

**Pull request merged**